backend/infrahub/graphql/mutations/branch.py

```diff
@@ -183,85 +183,22 @@ class Arguments:
         context: GraphqlContext = info.context
 
         if not context.service:
             raise ValueError("Service must be provided to rebase a branch.")
 
-        async with UserTask.from_graphql_context(title=f"Rebase branch : {data.name}", context=context) as task:
-            obj = await Branch.get_by_name(db=context.db, name=str(data.name))
-            merger = BranchMerger(db=context.db, source_branch=obj, service=context.service)
-
-            # If there are some changes related to the schema between this branch and main, we need to
-            #  - Run all the validations to ensure everything if correct before rebasing the branch
-            #  - Run all the migrations after the rebase
-            if obj.has_schema_changes:
-                candidate_schema = merger.get_candidate_schema()
-                constraints = await merger.calculate_validations(target_schema=candidate_schema)
-                error_messages, _ = await schema_validators_checker(
-                    branch=obj, schema=candidate_schema, constraints=constraints, service=context.service
-                )
-                if error_messages:
-                    raise ValidationError(",\n".join(error_messages))
-
-            schema_in_main_before = merger.destination_schema.duplicate()
-
-            async with context.db.start_transaction() as dbt:
-                await obj.rebase(db=dbt)
-                await task.info(message="Branch successfully rebased", db=dbt)
-
-            if obj.has_schema_changes:
-                # NOTE there is a bit additional work in order to calculate a proper diff that will
-                # allow us to pull only the part of the schema that has changed, for now the safest option is to pull
-                # Everything
-                # schema_diff = await merger.has_schema_changes()
-                updated_schema = await registry.schema.load_schema_from_db(
-                    db=context.db,
-                    branch=obj,
-                    # schema=merger.source_schema.duplicate(),
-                    # schema_diff=schema_diff,
-                )
-                registry.schema.set_schema_branch(name=obj.name, schema=updated_schema)
-                obj.update_schema_hash()
-                await obj.save(db=context.db)
-
-                # Execute the migrations
-                migrations = await merger.calculate_migrations(target_schema=updated_schema)
-
-                errors = await schema_migrations_runner(
-                    branch=merger.source_branch,
-                    new_schema=candidate_schema,
-                    previous_schema=schema_in_main_before,
-                    migrations=migrations,
-                    service=context.service,
-                )
-                for error in errors:
-                    context.service.log.error(error)
+        obj = await Branch.get_by_name(db=context.db, name=str(data.name))
```

Comment:
This line is duplicated line 197

Reply:
Both are required
the one validates that the branch is valid and since the workflow will change some values within the object, we need to re-requery it from the database after BRANCH_REBASE has been executed